source/Octopus.Client/OctopusClient.cs

```diff
@@ -648,14 +701,42 @@ protected virtual OctopusResponse<TResponseResource> DispatchRequest<TResponseRe
                     {
                         webResponse.Close();
                     }
-                        // ReSharper disable once EmptyGeneralCatchClause
+                    // ReSharper disable once EmptyGeneralCatchClause
                     catch
                     {
                     }
                 }
             }
         }
 
+        private void ValidateSpaceId(string spaceId)
+        {
+            if (string.IsNullOrEmpty(spaceId))
+            {
+                throw new ArgumentException("spaceId cannot be null");
+            }
+            if (spaceId == "all")
```

Comment:
same as above, put this definition somewhere similar to how we have `DefaultIdentifiers` in Octopus server